components/UserAuth/RequestReset.tsx

```diff
@@ -13,46 +19,68 @@ const REQUEST_RESET_MUTATION = gql`
   }
 `
 
+//so apparently it's a security issue if someone can test if an email is the database or not,
+//hence keystone is not checking for valid email.
+//i dont have the heart to delete this yet :D maybe it will come handy
+
+// const FIND_USER_BY_EMAIL_QUERY = gql`
+//   query FIND_USER_BY_EMAIL_QUERY($email: String) {
+//     allUsers(where: { email: $email }) {
+//       id
+//     }
+//   }
+// `
+
 export default function RequestReset() {
   const { inputs, handleChange, resetForm } = useForm({
     email: '',
   })
 
-  const [requestReset, { data, loading, error }] = useMutation(
+  const [requestReset, { data, loading }] = useMutation(
     REQUEST_RESET_MUTATION,
     {
       variables: inputs,
     },
   )
 
-  const handleSubmit = async (event) => {
+  // const [findEmail, { data: userData }] = useLazyQuery(
+  //   FIND_USER_BY_EMAIL_QUERY,
+  //   {
+  //     variables: inputs,
+  //   },
+  // )
+
+  // const error =
+  //   userData?.allUsers.length === 0
+  //     ? 'This email address is not in our database'
+  //     : undefined
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault()
-    await requestReset().catch(console.error)
+    await requestReset()
     resetForm()
   }
 
   return (
-    <>
-      <h2>Request password reset</h2>
-      <form method="POST" onSubmit={handleSubmit}>
-        <fieldset disabled={loading}>
-          {data?.sendUserPasswordResetLink === null && (
-            <p>Success! Check your email for a link!</p>
-          )}
-          <label htmlFor="email">
-            Email
-            <input
-              type="email"
-              name="email"
-              placeholder="Email address"
-              autoComplete="email"
-              value={inputs.email}
-              onChange={handleChange}
-            />
-          </label>
-          <button type="submit">Request reset</button>
-        </fieldset>
-      </form>
-    </>
+    <FormStyle method="POST" onSubmit={handleSubmit}>
+      <FormHeader>Request password reset</FormHeader>
+      <FieldSetStyle disabled={loading}>
+        {data?.sendUserPasswordResetLink === null && (
```

Comment:
`null` is a false value... or are you checking specific for `null`? That means `false` or `undefined` is not catched here

Reply:
this is an graphql specific, the requestReset will always result in null regardless if successful or not